### DISCUSSION TITLE
What do you guys use for UI tests on a CI/CD pipeline?

### DISCUSSION THREAD

**Discussion by Pedroo123** [MobileNativeFoundation/discussions#216] [General]
To give some context, i have a QA team that has a pretty good amount of test cases already automated, using java, appium, and the good old selenium for our native Android app.
But, at the moment we just run them on demand, on our local environment (desktops or laptops).
What im doing right now, is building a dockerfile to emulate a device, and then, i might be able to run all these tests cases on a scheduled and zero cost CI/CD pipeline.

Am i doing some overkill here? To be fair, my organization is actually VERY big, a hermetic emulator would be a pretty good solution for my problem, but i dont know if it escalates on a decent level...
So yeah, im open to advices or even read sucess cases using tools like paid device farms or even open source tools :)

**Comment by esoxjem** [comment]
@Pedroo123 try Maestro
https://maestro.mobile.dev/

**Reply by Pedroo123** [reply]
I'll check it out, never heard of it TBH xD
